odoo/tools/mail.py

```diff
@@ -667,6 +667,7 @@ def formataddr(pair, charset='utf-8'):
         domain = idna.encode(domain).decode('ascii')
 
     if name:
+        name = name.replace('@', '-')
```

Comment:
Pinging @Julien00859 for a last review, do you foresee issues with this kind of change ?

@datonefaridze Would glady have a comment in code explaining why you do that (otherwise people landing on that code will probably scratch their heads).

Reply:
Frankly I only see benefits!

I think a comment is not the correct place to document this, I would move it inside the docstring:
```diff
diff --git a/odoo/tools/mail.py b/odoo/tools/mail.py
index 4f0089520f79..601c234fe88e 100644
--- a/odoo/tools/mail.py
+++ b/odoo/tools/mail.py
@@ -692,6 +692,9 @@ def formataddr(pair, charset='utf-8'):
     If the first element of pair is falsy then only the email address
     is returned.
 
+    All "@" characters found in ``realname`` are replaced by "-" for
+    better compatibility with email softwares (including ours).
+
     Set the charset to ascii to get a RFC-2822 compliant email. The
     realname will be base64 encoded (if necessary) and the domain part
     of the email will be punycode encoded (if necessary). The local part
```
I find the comment overly verbose, IMO people can just look up the commit message.

Reply:
@Julien00859 done